src/main/java/com/example/bookstore/controller/admin/BooksAdminController.java

```diff
@@ -0,0 +1,62 @@
+package com.example.bookstore.controller.admin;
+
+import com.example.bookstore.entity.Book;
+import com.example.bookstore.entity.BookCategory;
+import com.example.bookstore.entity.Category;
+import com.example.bookstore.service.BookCategoriesService;
+import com.example.bookstore.service.BooksService;
+import com.example.bookstore.service.CategoriesService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+
+@Controller
+@RequestMapping("/admin/books")
+public class BooksAdminController {
```

Comment:
```suggestion
public class BooksController {
```
thêm chữ admin là j, có package admin dc rồi

Reply:
Nó k cho chị ơi, kể cả có pakage admin nó vẫn báo lỗi chị ạ